src/Raven.Client/Server/DatabaseTopology.cs

```diff
@@ -165,23 +165,21 @@ public static (List<ReplicationNode> addDestinations, List<ReplicationNode> remo
                 while (hasNewValues && hasOldValues)
                 {
                     var res = oldEnum.Current.CompareTo(newEnum.Current);
-                    switch (res)
+                    if (res > 0)
                     {
-                        case 1:
-                            addDestinations.Add(newEnum.Current);
-                            hasNewValues = newEnum.MoveNext();
-                            break;
-                        case -1:
-                            removeDestinations.Add(oldEnum.Current);
-                            hasOldValues = oldEnum.MoveNext();
-                            break;
-                        case 0:
-                            hasNewValues = newEnum.MoveNext();
-                            hasOldValues = oldEnum.MoveNext();
-                            break;
-                        default:// should never happen
-                            throw new InvalidDataException($"{res} is an invalid comparison result between {oldEnum.Current.Humane} and {newEnum.Current.Humane}");
+                        addDestinations.Add(newEnum.Current);
+                        hasNewValues = newEnum.MoveNext();
+                        break;
```

Comment:
breaks where there in the switch cases.. missed it
moved from switch to if-cases because while debugging i got an exception there when compareTo returned 2